__tests__/core/gatherer.test.ts

```diff
@@ -54,4 +63,84 @@ describe('Gatherer', () => {
     expect(network.start).toHaveBeenCalled();
     await Gatherer.stop();
   });
+
+  describe('Append Elastic/Synthetics to UA', () => {
+    it('works on a single page', async () => {
+      const driver = await Gatherer.setupDriver({ wsEndpoint });
+
+      expect(await driver.page.evaluate(() => navigator.userAgent)).toContain(
+        ' Elastic/Synthetics'
+      );
+
+      await Gatherer.dispose(driver);
+      await Gatherer.stop();
+    });
+
+    it('works with device emulation', async () => {
+      const driver = await Gatherer.setupDriver({
+        wsEndpoint,
+        playwrightOptions: { ...devices['Galaxy S9+'] },
+      });
+
+      expect(await driver.page.evaluate(() => navigator.userAgent)).toContain(
+        ' Elastic/Synthetics'
+      );
+
+      await Gatherer.dispose(driver);
+      await Gatherer.stop();
+    });
+
+    it('works with popup window', async () => {
+      const driver = await Gatherer.setupDriver({
+        wsEndpoint,
+        playwrightOptions: { ...devices['Galaxy S9+'] },
+      });
+
+      const { page, context } = driver;
+
+      await page.goto(server.TEST_PAGE);
+      await page.setContent(
+        '<a target=_blank rel=noopener href="/popup.html">popup</a>'
+      );
+
+      const [page1] = await Promise.all([
+        context.waitForEvent('page'),
+        page.click('a'),
+      ]);
+      await page1.waitForLoadState();
+
+      expect(await page1.evaluate(() => navigator.userAgent)).toContain(
+        ' Elastic/Synthetics'
+      );
+
+      expect(await page1.textContent('body')).toEqual('Not found');
+
+      await Gatherer.dispose(driver);
+      await Gatherer.stop();
+    });
+
+    it('works with iframe content', async () => {
```

Comment:
I would remove this iframe test in favor of request test. Or merge it on the previous popup test to look for request to `popup.html` and check if request.headers.userAgent contains the Elastic identifier. 

Iframes tests are unnecessary as they inherit the parent document.